library/common.h

```diff
@@ -191,21 +191,29 @@ inline void mbedtls_xor(unsigned char *r, const unsigned char *a, const unsigned
         uint8x16_t x = veorq_u8(v1, v2);
         vst1q_u8(r + i, x);
     }
+    if (n % 16 != 0)
 #elif defined(MBEDTLS_ARCH_IS_X64) || defined(MBEDTLS_ARCH_IS_ARM64)
     /* This codepath probably only makes sense on architectures with 64-bit registers */
     for (; (i + 8) <= n; i += 8) {
         uint64_t x = mbedtls_get_unaligned_uint64(a + i) ^ mbedtls_get_unaligned_uint64(b + i);
         mbedtls_put_unaligned_uint64(r + i, x);
     }
+    // This if statement helps some compilers (e.g., IAR) optimise out the byte-by-byte tail case
+    // where n is a constant multiple of 8.
+    // It makes no difference for others (e.g. recent gcc and clang).
+    if (n % 8 != 0)
```

Comment:
Would it be better if this comment was on the first instance of `if (n % N) != 0)`?